### PR TITLE
[core] Add HTTPS support

### PR DESCRIPTION
**Pull request by raylu**
We have collectors deployed behind an AWS NLB that does TLS termination so we'd like HTTPS support.

This will have conflicts with #1054.

### CODE REVIEW

**Comment by raylu** [comment]
rebasing on top of what got merged this morning (LMK if this is not necessary)

**Comment by raylu** [comment]
(@berkowitzi fixed passing the `agent_https` config through so we could actually turn it on)

File: tests/test_api.py [hunk 1]
```diff
@@ -111,10 +111,10 @@ def read(self):
 
 
 def test_api_str():
-    api = API('localhost', 8126)
-    assert str(api) == 'localhost:8126'
+    api = API('localhost', 8126, https=True)
+    assert str(api) == 'https://localhost:8126'
     api = API('localhost', 8126, '/path/to/uds')
-    assert str(api) == '/path/to/uds'
+    assert str(api) == 'http:///path/to/uds'
```
Comment (skaji) [inline tests/test_api.py]: This should be `unix:///path/to/uds`?

Comment (raylu) [reply tests/test_api.py]: that was my initial plan, but then how do you signify HTTPS + UDS? should it just not be allowed?

Comment (skaji) [reply tests/test_api.py]: https means HTTP over TLS. AFAIK, TLS runs normally on TCP.
So I don't think we should care about Unix Domain Socket + TLS.

By the way, `unix:///path/to/uds` is not accurate. In fact, we speak HTTP over Unix Domain Socket. So we should probably write `http+unix` or something.
`unix:///path/to/uds` is an abbreviated notation.

Comment (skaji) [reply tests/test_api.py]: I just want to make sure that this PR is consistent with https://github.com/DataDog/dd-trace-py/pull/1054.

Comment (raylu) [reply tests/test_api.py]: ok. I went with `unix://` for both HTTP and HTTPS

File: ddtrace/internal/writer.py [hunk 2]
```diff
@@ -193,8 +193,7 @@ def put(self, item):
             Queue.put(self, item, block=False)
         except Full:
             # If the queue is full, replace a random item. We need to make sure
-            # the queue is not emptied was emptied in the meantime, so we lock
-            # check qsize value.
+            # the queue was not in the meantime, so we lock and check qsize.
```
[the remark this answers is not in the document]
Comment (raylu) [reply ddtrace/internal/writer.py]: undid this change

**Comment by brettlangdon** [comment]
I merged down from `master` so we'll wait on tests to pass, and for @jd to give one last look through.

**Pull request merged**